SDWebImage/SDWebImagePrefetcher.m

```diff
@@ -256,12 +261,20 @@ @implementation SDWebImagePrefetchToken
 
 - (void)cancel {
     @synchronized (self) {
-        for (id operation in self.operations) {
+        [self.prefetchOperations compact];
+        for (id operation in self.prefetchOperations) {
+            SDAsyncBlockOperation *asyncBlockOperation = operation;
```

Comment:
This one can use the code below, since all `NSOperation` conforms to `SDWebImageOperation` protocol

```
if ([operation conformsToProtocol:@protocol(SDWebImageOperation)]) {
    [operation cancel];
}
```

Reply:
Oh, I see, I can consolidate them.

Reply:
。。。Wait. Don't change this line of code. Change the below one into this type.
Previous there are no `compat` call, so the for in loop element can be a `NULL`. But actually, these code is usless because the element type of NSPointerArray is fixed.

Reply:
... What's your mean? Is this cause issue?

Reply:
😅 Both of them is OK...I means we can use the same code for this two weak array. Choose what you like.

You can also write:

```objective-c
[self.prefetchOperations compact];
for (id operation in self.prefetchOperations) {
    SDAsyncBlockOperation *prefetchOperation = operation;
    [prefetchOperation cancel];
}
self.prefetchOperations.count = 0;

[self.loadOperations compact];
for (id operation in self.loadOperations) {
    SDWebImageCombinedOperation *loadOperation = operation;
    [loadOperation cancel];
}
self.loadOperations.count = 0;
```